### ISSUE TITLE
Support after/deferred property callbacks when creating VDOM

### ISSUE THREAD

**Issue by pottedmeat** [dojo/widget-core#714]
**Enhancement** 

Some VDOM properties cannot be calculated before DOM has been rendered because they depend on values generated as part of the render. We should support a method of supplying a function to be called immediately after these values can be read once a render is complete. The returned value of this function would be applied to the rendered node.

**Comment by agubler** [comment]
@pottedmeat could you please add your specific use case for deferred property callbacks?

**Comment by pottedmeat** [comment]
For example, in some browsers `scrollTop` is automatically updated as content is added or removed "above the fold" inside a scrolling node. In others, it is not. To manually update this value, we need to know the size of the removed content and the size of the added content as well as where it is.

**Comment by kitsonk** [comment]
With all the changes is this use case still valid?  If so, we should figure out what work to do, or otherwise close this issue.

**Comment by matt-gadd** [comment]
@kitsonk yup these are still valid, and we should now be able to implement given we have control of the vdom lifecycle.
